src/main/java/de/tum/in/www1/artemis/service/connectors/localci/LocalCISharedBuildJobQueueService.java

```diff
@@ -461,6 +467,14 @@ private void processBuild(LocalCIBuildJobQueueItem buildJob) {
         });
     }
 
+    private void addToRecentBuildJobs(LocalCIBuildJobQueueItem updatedJob) {
+        if (recentBuildJobs.size() >= 5) {
+            recentBuildJobs.remove(0);
+        }
+        recentBuildJobs.add(updatedJob);
+        updateLocalBuildAgentInformation();
+    }
```

Comment:
Please add a comment explaining why the remove call is necessary. Maybe consider moving this magic number into a named constant.